Userland/Services/WindowServer/Screen.cpp

```diff
@@ -462,7 +462,10 @@ void ScreenInput::on_receive_mouse_data(const MousePacket& packet)
     }
 
     if (packet.z) {
-        auto message = make<MouseEvent>(Event::MouseWheel, m_cursor_location, buttons, MouseButton::None, m_modifiers, packet.z * m_scroll_step_size);
+        auto message = make<MouseEvent>(Event::MouseWheel, m_cursor_location, buttons, MouseButton::None, m_modifiers, packet.z * m_scroll_step_size, WheelDirection::Vertical);
+        Core::EventLoop::current().post_event(WindowManager::the(), move(message));
+    } else if (packet.w) {
+        auto message = make<MouseEvent>(Event::MouseWheel, m_cursor_location, buttons, MouseButton::None, m_modifiers, packet.w * m_scroll_step_size, WheelDirection::Horizontal);
         Core::EventLoop::current().post_event(WindowManager::the(), move(message));
```

Comment:
As this is common code, it may be cleaner to move it outside of the conditionals (even if you have to work around some stuff with the variable).